src/DateTimeParser.jl

```diff
@@ -524,27 +547,28 @@ end
 
 "Converts a 2 digit year to a 4 digit one within 50 years of convert_year. At the momment
  convert_year defaults to 2000, if people are still using 2 digit years after year 2049
- (hopefully not) then we can change the default to year(today())"
-function convertyear(year::Int, convert_year=2000)
-    if year <= 99
-        century = convert_year - convert_year % 100
-        year += century
-        if abs(year - convert_year) >= 50
-            if year < convert_year
-                year += 100
+ (hopefully not) then we can change the default to Year(today())"
+function convertyear(year::Year, convert_year=Year(2000))
+    value = year.value
+    if value <= 99
+        century = convert_year.value - (convert_year.value % 100)
+        value += century
+        if abs(value - convert_year.value) >= 50
+            if value < convert_year.value
+                value += 100
             else
-                year -= 100
+                value -= 100
             end
         end
     end
-    return year
+    return Year(value)
 end
 
-function converthour(hour::Int, ampm::Symbol)
-    if hour < 12 && ampm == :pm
-        hour += 12
-    elseif hour == 12 && ampm == :am
-        hour = 0
+function converthour(hour::Hour, ampm::Symbol)
+    if hour.value < 12 && ampm == :pm
+        hour = Hour(hour.value + 12)
+    elseif hour.value == 12 && ampm == :am
+        hour = Hour(0)
     end
     return hour
 end
```

Comment:
You could switch these functions back to using Int's and then have these new methods:
```
import Base.Dates: value
converthour(hour::Hour, ampm::Symbol) = Hour(converthour(value(hour)))
convertyear(year::Year, convert_year=Year(2000)) = Year(convertyear(value(year), value(convert_year)))
```